tasks/create_master_user.yml

```diff
@@ -0,0 +1,134 @@
+---
+- name: "List existing users"
+  uri:
+    url: "{{ gitlab_external_url }}/api/{{ gitlab_api_version }}/users"
+    method: "GET"
+    headers:
+      PRIVATE-TOKEN: "{{ gitlab_admin_token }}"
+    validate_certs: "{{ gitlab_validate_certs }}"
+    return_content: True
+  register: user_list_output
+
+- name: "Check if user is already exists"
+  set_fact:
+    user_exists: "{{ gitlab_master_username in (user_list_output.json |\
+      map(attribute='username') | list) }}"
+
+- name: "Create user"
+  uri:
+    url: "{{ gitlab_external_url }}/api/{{ gitlab_api_version }}/users"
+    method: "POST"
+    headers:
+      PRIVATE-TOKEN: "{{ gitlab_admin_token }}"
+    body: "name={{ gitlab_master_name }}&username={{ gitlab_master_username }}&\
+      email={{ gitlab_master_email }}&password={{ gitlab_master_password }}&\
+      skip_confirmation=true"
+    body_format: raw
+    validate_certs: "{{ gitlab_validate_certs }}"
+    return_content: True
+    status_code: 201
+  register: user_output
+  when: not user_exists
+
+- name: "Set user ID"
+  set_fact:
+    gitlab_master_id: "{{ user_output.json.id }}"
+  when: not user_exists
+
+- name: "Get user ID"
+  set_fact:
+    gitlab_master_id: "{{ item.id }}"
+  loop: "{{ user_list_output.json }}"
+  when:
+    - user_exists
+    - item.username == gitlab_master_username
+
+- name: "User debug"
+  debug:
+    msg: "User id is {{ gitlab_master_id }}"
```

Comment:
We still need it?

Reply:
yes - we need it to get the ID of created user, group and project for further needs